packages/web-components/fast-ssr/src/template-parser/template-parser.ts

```diff
@@ -0,0 +1,351 @@
+}
+
+/**
+ * Tests if a node is an interpolated FAST marker
+ * @param node - the node to test
+ */
+function isInterpolationMarkerNode(
+    node: DefaultTreeNode
+): node is Required<DefaultTreeTextNode> {
+    return (
+        isTextNode(node) &&
+        node.sourceCodeLocation !== undefined &&
+        isInterpolationMarker(node)
+    );
+}
+
+/**
+ * Tests if a node is a FAST comment marker
+ * @param node - the node to test
+ */
+function isCommentMarkerNode(
+    node: DefaultTreeNode
+): node is Required<DefaultTreeCommentNode> {
+    return (
+        isCommentNode(node) &&
+        node.sourceCodeLocation !== undefined &&
+        isMarkerComment(node)
+    );
+}
+
+/**
+ * Parses a template into a set of operation instructions
+ * @param template - The template to parse
+ */
+export function parseTemplateToOpCodes(template: ViewTemplate): Op[] {
+    const cached: Op[] | undefined = opCache.get(template);
+    if (cached !== undefined) {
+        return cached;
+    }
+
+    const { html } = template;
+
+    if (typeof html !== "string") {
+        throw new Error(
+            "@microsoft/fast-ssr only supports rendering a ViewTemplate with a string source."
+        );
+    }
+
+    /**
+     * Typescript thinks that `html` is a string | HTMLTemplateElement inside the functions defined
+     * below, so store in a new var that is just a string type
+     */
+    const templateString = html;
+    const nodeTree = parseFragment(html, { sourceCodeLocationInfo: true });
+
+    if (!("nodeName" in nodeTree)) {
+        // I'm not sure when exactly this is encountered but the type system seems to say it's possible.
+        throw new Error(`Error parsing template:\n${template}`);
+    }
+
+    /**
+     * Tracks the offset location in the source template string where the last
+     * flushing / skip took place.
+     */
+    let lastOffset: number | undefined = 0;
+
+    /**
+     * Collection of op codes
+     */
+    const opCodes: Op[] = [];
+    opCache.set(template, opCodes);
+
+    const { directives } = template;
+
+    /**
+     * Parses an Element node, pushing all op codes for the element into
+     * the collection of ops for the template
+     * @param node - The element node to parse
+     */
+    function parseElementNode(node: DefaultTreeElement): void {
+        // Track whether the opening tag of an element should be augmented.
+        // All constructable custom elements will need to be augmented,
+        // as well as any element with attribute bindings
+        let augmentOpeningTag = false;
+        const { tagName } = node;
+        let ctor: typeof HTMLElement | undefined;
+
+        // Sort attributes by whether they're related to a binding or if they have
+        // static value
+        const attributes: {
+            static: Map<string, string>;
+            dynamic: Attribute[];
+        } = node.attrs.reduce(
+            (prev, current) => {
+                if (isInterpolationMarker(current)) {
```

Comment:
Here you could use the `Parser` API. If it returns null, then there is no interpolation. If it returns anything else, you could push that into the dynamic array for use later.